### PR TITLE
add heuristic_cache and softmax_dropout cpp benchmarks to cmakelist

### PR DESCRIPTION
**Pull request by liqiangxl**
I'm not sure why these two C++ benchmarks are not being listed to compile in the cmakelist. They are not in the cmakelist since the initial commit of this repo. `softmax_dropout` is helpful to test RNG ops, so I added them back.
(1) adjusted file orders in CMakeLists.txt
(2) minor code changes in `heuristic_cache` and `softmax_dropout` to fix compile errors.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

File: CMakeLists.txt [hunk 1]
```diff
@@ -514,28 +514,30 @@ if(BUILD_NVFUSER_BENCHMARK)
...
-    ${NVFUSER_ROOT}/benchmark/softmax.cpp
-    ${NVFUSER_ROOT}/benchmark/softmax_backward.cpp
+    ${NVFUSER_ROOT}/benchmark/rms_norm_backward.cpp
+    ${NVFUSER_ROOT}/benchmark/rms_norm.cpp
     ${NVFUSER_ROOT}/benchmark/scale_bias_relu.cpp
-    ${NVFUSER_ROOT}/benchmark/transpose.cpp
-    ${NVFUSER_ROOT}/benchmark/matmul.cpp
+    ${NVFUSER_ROOT}/benchmark/shape_inference.cpp
+    ${NVFUSER_ROOT}/benchmark/softmax_backward.cpp
+    ${NVFUSER_ROOT}/benchmark/softmax_dropout.cpp
```
Comment (jjsjann123) [inline CMakeLists.txt:536]: so we missed `softmax_dropout.cpp` and `heuristic_cache.cpp` the whole time :laughing:

File: benchmark/heuristic_cache.cpp [hunk 2]
```diff
@@ -76,8 +76,8 @@ static auto getLayerBackwardNormRuntime(
   at::Tensor aten_input = at::randn(shape, options);
   at::Tensor aten_weight = at::randn(norm_shape, options);
   at::Tensor aten_bias = at::randn(norm_shape, options);
-  auto at_weight = std::optional<at::Tensor>(aten_weight);
-  auto at_bias = std::optional<at::Tensor>(aten_bias);
+  auto at_weight = c10::optional<at::Tensor>(aten_weight);
+  auto at_bias = c10::optional<at::Tensor>(aten_bias);
```
Comment (jjsjann123) [inline benchmark/heuristic_cache.cpp:80]: Do we need to wrap things explicitly here? Can't we just use `aten_weight`/`aten_bias` in places of `at_weight`/`at_bias` below?

**Review by jjsjann123 — APPROVED** [review]
Thx for clean up.

File: benchmark/heuristic_cache.cpp (hunk 2 above)
Comment (liqiangxl) [reply benchmark/heuristic_cache.cpp]: It also works! changed.

File: CMakeLists.txt (hunk 1 above)
Comment (liqiangxl) [reply CMakeLists.txt:535]: that's right. I was doing some benchmarks on RNG ops and realized `NvFuserScheduler_Softmax_Dropout*` are not compiled at all.

**Pull request merged**